plugins/needs_restarting.py

```diff
@@ -40,6 +40,113 @@
 NEED_REBOOT = ['kernel', 'glibc', 'linux-firmware', 'systemd', 'dbus',
                'dbus-broker', 'dbus-daemon']
 
+class DirectoryOption(object):
+    """ This class handles a directory containing file options
+        This class can sanitize dir path
+    """
+    def __init__(self, path=None):
+        self._path = self._parse(path)
+
+    def _parse(self, path):
+        if not isinstance(path, str):
+            return set(path)
+        strval = path.replace('\n', ' ')
+        strval = path.replace(',', ' ')
+        strval = { x for x in strval.split(' ') if x }
+        strval = { x for x in strval if self._is_dir(x) }
+        return strval
+
+    def _is_dir(self, path=None):
+        return os.path.isdir(path)
+
+    #def _is_empty(self):
+    #    if len(os.listdir(self._path) ) == 0:
+    #        return False
+    #    return True
+
+    @property
+    def path(self):
+        return self._path
```

Comment:
Naming the variable `path` is actually quite confusing, since it's a set of paths.